packages/ui/src/machines/authenticator/defaultServices.ts

```diff
@@ -14,13 +14,35 @@ export const defaultServices = {
   // Validation hooks for overriding
   async validateCustomSignUp(formData): Promise<ValidatorResult> {},
   async validateConfirmPassword<Validator>(formData): Promise<ValidatorResult> {
-    const { password, confirm_password } = formData;
+    const {
+      password,
+      confirm_password,
+      touched_confirm_password,
+      touched_password,
+    } = formData;
 
     if (!password && !confirm_password) {
       // these inputs are clean, don't complain yet
       return null;
-    } else if (password && confirm_password && password !== confirm_password) {
-      // only return an error if both fields have text entered and the passwords do not match
+    } else if (
+      (password || confirm_password) &&
+      touched_password &&
+      touched_confirm_password &&
+      password !== confirm_password
+    ) {
+      return {
+        confirm_password: 'Your passwords must match',
```

Comment:
New check if they've entered text in either `password` or `confirm_password` and they've touched both, and the passwords don't match, then show error.